src/main/java/core/basesyntax/figures/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax.figures.suppliers;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURES_COUNT = 5;
+    private static final int DEFAULT_RADIUS = 10;
+    private static final String DEFAULT_COLOR = "white";
+
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figure figure;
+
+        int figureNumber = random.nextInt(FIGURES_COUNT);
+        String randomColor = colorSupplier.getRandomColor();
+
+        switch (figureNumber) {
+            case 0:
+                return getDefaultFigure();
+            case 1:
+                figure = new IsoscelesTrapezoid(getRandomGeometricProperty(),
+                        getRandomGeometricProperty(),
+                        getRandomGeometricProperty(), randomColor);
+                break;
+            case 2:
+                figure = new Rectangle(getRandomGeometricProperty(),
+                        getRandomGeometricProperty(), randomColor);
+                break;
+            case 3:
+                figure = new RightTriangle(getRandomGeometricProperty(),
+                        getRandomGeometricProperty(), randomColor);
+                break;
+            default:
+                figure = new Square(getRandomGeometricProperty(), randomColor);
+        }
+        return figure;
```

Comment:
```suggestion
        int figureNumber = random.nextInt(FIGURES_COUNT);
        String randomColor = colorSupplier.getRandomColor();

        return switch (figureNumber) {
            case 0:
                yield getDefaultFigure();
            case 1:
                yield new IsoscelesTrapezoid(getRandomGeometricProperty(),
                        getRandomGeometricProperty(),
                        getRandomGeometricProperty(), randomColor);
            case 2:
                yield new Rectangle(getRandomGeometricProperty(),
                        getRandomGeometricProperty(), randomColor);
            case 3:
                yield new RightTriangle(getRandomGeometricProperty(),
                        getRandomGeometricProperty(), randomColor);
            default:
                yield new Square(getRandomGeometricProperty(), randomColor);
        };
```

Reply:
let's use newer java features and simplify this code removing unnecessary variable assignment

Reply:
![2024-01-17_17-13-38](https://github.com/mate-academy/jv-oop-advanced/assets/108012785/f304f321-f749-4bf1-95c9-300533d2f87d)

Build Failure